ui/org.eclipse.pde.launching/src/org/eclipse/pde/launching/AbstractPDELaunchConfiguration.java

```diff
@@ -155,21 +155,25 @@ private String[] updateVMArgumentWithAdditionalArguments(String[] args, boolean
 			addModuleSystem = true;
 			argLength++; // Need to add the argument
 		}
-		IVMInstall vmInstall;
-		try {
-			vmInstall = VMHelper.getVMInstall(configuration);
-			if (vmInstall instanceof AbstractVMInstall) {
-				AbstractVMInstall install = (AbstractVMInstall) vmInstall;
-				String vmver = install.getJavaVersion();
-				if (vmver != null && JavaCore.compareJavaVersions(vmver, JavaCore.VERSION_17) >= 0) {
-					if (!argumentContainsAttribute(args, allowSecurityManager)) {
-						addAllowSecurityManager = true;
-						argLength++; // Need to add the argument
+		final String buildId = System.getProperty("eclipse.buildId", "Unknown Build"); //$NON-NLS-1$//$NON-NLS-2$
```

Comment:
I don't know the details how this class actually works but to me it looks like you are looking at the build ID of the Eclipse IDE that launches the application. However, the crucial part of the Eclipse version of the *launched* application, i.e. the one in the target platform. And that can be different from the currently running platform.

Reply:
Ok, makes sense. will work on that.